packages/components/pie-notification/src/index.ts

```diff
@@ -202,25 +240,103 @@ export class PieNotification extends LitElement implements NotificationProps {
                 size="small"
                 class="${componentClass}-icon-close"
                 data-test-id="${componentSelector}-icon-close"
+                @click="${this.handleCloseButton}"
                 >
                 <icon-close></icon-close>
             </pie-icon-button>`;
     }
 
+    /**
+     * It handles the action when user clicks in the close button.
+     * Also triggers an event when is executed.
+     *
+     * @private
+     */
+    private handleCloseButton () {
+        this.closeNotificationComponent();
+        dispatchCustomEvent(this, ON_NOTIFICATION_CLOSE_EVENT, { targetNotification: this });
+    }
+
+    /**
+     * Util method responsible to close the component.
+     *
+     * @private
+     */
+    private closeNotificationComponent () {
+        this.isOpen = false;
+    }
+
+    /**
+     * It handle the action button action. It is called if action button has `onClick` as part of its action props.
+     * Also triggers an event according to its `actionType`.
+     *
+     * @param {NonNullable<ActionProps['onClick']>} action
+     * @param {'leading' | 'supporting'} actionType
+     *
+     * @private
+     */
+    private handleActionClick (action: NonNullable<ActionProps['onClick']>, actionType: 'leading' | 'supporting') {
+        const EVENT = actionType === 'leading' ? ON_NOTIFICATION_LEADING_ACTION_CLICK_EVENT : ON_NOTIFICATION_SUPPORTING_ACTION_CLICK_EVENT;
+
+        action();
+
+        dispatchCustomEvent(this, EVENT, { targetNotification: this });
+    }
+
+    /**
+     * Render the action button depending on action type and its action.
+     *
+     * @param {ActionProps} action - The action properties.
+     * @param {'leading' | 'supporting'} actionType - The type of the action.
+     *
+     * @returns {TemplateResult | typeof nothing} - The rendered action button or nothing if the action text is not defined.
+     * @private
+     */
+    private renderActionButton (action: ActionProps, actionType: 'leading' | 'supporting') : TemplateResult | typeof nothing {
+        const { text, ariaLabel, onClick } = action;
+
+        if (!text) {
+            return nothing;
+        }
+
+        const buttonVariant = actionType === 'leading' ? 'primary' : 'ghost';
+
+        return html`
+            <pie-button
+                variant="${buttonVariant}"
+                size="small-productive"
+                aria-label="${ariaLabel || nothing}"
+                @click="${onClick ? () => this.handleActionClick(onClick, actionType) : nothing}"
+                data-test-id="${componentSelector}-${actionType}-action"
+                ?isFullWidth="${this.hasStackedActions}"
+                type="submit">
```

Comment:
i think we should use type="submit" here since the parent element isn't a form. 
do you think we can use type="button" instead?

Reply:
You're totally right!